.github/workflows/ci.yml

```diff
@@ -0,0 +1,39 @@
+# build-avr-gcc
+#
+# Copyright 2019, 2021, Andrew Countryman <apcountryman@gmail.com> and the build-avr-gcc
+# contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+# file except in compliance with the License. You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# File: .github/workflows/ci.yml
+# Description: GitHub Actions CI workflow for the build-avr-gcc repository.
+
+name: CI
+
+on:
+    push:
+        branches:
+            - main
+    pull_request:
+        branches:
+            - main
+
+jobs:
+    analyze-static-shellcheck:
+        name: Analyze (static) - shellcheck
+        runs-on: ubuntu-20.04
+        steps:
+            - uses: actions/checkout@v2
+              with:
+                  submodules: recursive
```

Comment:
Submodule checkout is not necessary since this project does not have any submodules.
```suggestion
```